core/domain/exp_domain_test.py

```diff
@@ -440,25 +441,28 @@ def test_validation(self):
 
         # A terminal interaction without a default outcome or answer group is
         # valid. This resets the exploration back to a valid state.
-        interaction.answer_groups = []
+        init_state.update_interaction_answer_groups([])
         exploration.validate()
 
         # Restore a valid exploration.
         interaction.id = 'TextInput'
-        interaction.answer_groups = answer_groups
-        interaction.default_outcome = default_outcome
+        answer_groups_list = [
+            answer_group.to_dict() for answer_group in answer_groups]
+        init_state.update_interaction_answer_groups(answer_groups_list)
+        init_state.update_interaction_default_outcome(default_outcome.to_dict())
```

Comment:
This seems a bit odd. Shouldn't we be passing domain objects around, not dicts?

Reply:
But can we do the conversion first (in the caller) and then pass the domain object in? Generally no other method should accept dicts except from_dict().

Reply:
Hi @DubeySandeep -- wanted to flag this as an unresolved comment.